python/feathub/processors/spark/spark_processor.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2022 The Feathub Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from concurrent.futures import ThreadPoolExecutor
+from datetime import timedelta, datetime
+from typing import Union, Optional, Dict
+
+import pandas as pd
+from pyspark.sql import SparkSession
+
+from feathub.common.exceptions import FeathubException
+from feathub.dsl.expr_parser import ExprParser
+from feathub.feature_tables.feature_table import FeatureTable
+from feathub.feature_views.feature_view import FeatureView
+from feathub.processors.processor import Processor
+from feathub.processors.spark.ast_evaluator.spark_ast_evaluator import SparkAstEvaluator
+from feathub.processors.spark.dataframe_builder.spark_dataframe_builder import (
+    SparkDataFrameBuilder,
+)
+from feathub.processors.spark.spark_job import SparkJob
+from feathub.processors.spark.spark_processor_config import (
+    SparkProcessorConfig,
+    MASTER_CONFIG,
+)
+from feathub.processors.spark.spark_table import SparkTable
+from feathub.registries.registry import Registry
+from feathub.table.table_descriptor import TableDescriptor
+
+
+class SparkProcessor(Processor):
+    """
+    The SparkProcessor does feature ETL using Spark as the processing engine.
+
+    In the following we describe the keys accepted by the `config` dict passed to the
+    SparkProcessor constructor.
+
+    master: The Spark master URL to connect to.
+    """
+
+    def __init__(self, props: Dict, registry: Registry):
+        """
+        Instantiate the SparkProcessor.
+
+        :param props: The processor properties.
+        :param registry: An entity registry.
+        """
+        super().__init__()
+        self.registry = registry
+
+        config = SparkProcessorConfig(props)
+        spark_builder = SparkSession.builder
+
+        if config.get(MASTER_CONFIG) is not None:
```

Comment:
Would it be simpler to always require user to explicitly specify master URL?

Reply:
In Spark, `master` is an optional configuration that has a valid default value. I'm not sure we should implement a configuration different from that of Spark.

Reply:
Can you provide link to the doc that shows "master is an optional configuration"? It will be useful to check whether that doc explains the default behavior and make sure SparkProcessorConfig delivers at least the same readability/usability.

Reply:
If we remove these two lines of code, which makes the result look like this
```python
spark_session = SparkSession.builder.getOrCreate()
```
The SparkProcessor's test cases can still pass, which means the master configuration is not a necessity.

Apart from the phenomenon above, all of Spark's quickstart documents seem to explicitly set `master`. Spark's Configurations ([link](https://spark.apache.org/docs/latest/configuration.html)) document does not provide information about which configurations are optional.

Reply:
According to this doc, the value for `spark.master` is none, which means it is undefined what value is chosen if users does not explicitly specify value for `spark.master`. That means FeatHub should either require user to specify value for spark.master, or explicitly provide a value for it. We can not rely on the behavior of existing "undefined" Spark implementation.

[1] https://spark.apache.org/docs/latest/configuration.html

Reply:
Got it. I think it is better to require users to specify value for this configuration.